Refactoring of UI classes

TheMSB opened the pull request:
Converted whitespace usage to Google styled Java, rewtrote documentation and added where neccessary.

TheMSB commented:
I left the switch statement in Main alone since it seemed that applying Google style here would decrease readability of the code.

wandernauta commented:
Hey there!

Looks good overall. In a few places (you'll see them right away), you decided to break a call into two lines, like this:

````
dragStart = new Point2D(mouseEvent.getScreenX(),
         mouseEvent.getScreenY());
````

I'm personally not a great fan of this - I'd either keep it on one line (even if that pushes me ever so slightly over 100 characters) or refactor the arguments, like so:

````
double x = mouseEvent.getScreenX();
double y = mouseEvent.getScreenY();
dragStart = new Point2D(x, y);
````

The same goes for Javadoc: if it fits on one line, why burn two?

Still, if you're sure that this is the Right Way I'll merge right away, no questions asked :shipit:

TheMSB commented:
Which Class are you referring to? I agree if the difference is a few characters, in any case I'd like to check the lines.

wandernauta commented:
I'll mark them for you.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/CustomUIPane.java:
```diff
@@ -78,11 +79,13 @@ private void handleKey(KeyEvent keyEvent) {
         offset = new Point2D(this.getTranslateX(), this.getTranslateY());
-        dragStart = new Point2D(mouseEvent.getScreenX(), mouseEvent.getScreenY());
+        dragStart = new Point2D(mouseEvent.getScreenX(),
+                mouseEvent.getScreenY());
```
Keep as one line or refactor.

TheMSB replied:
Fixed

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/CustomUIPane.java:
```diff
@@ -78,11 +79,13 @@ private void handleKey(KeyEvent keyEvent) {
     private void handleDrag(MouseEvent mouseEvent) {
-        Point2D dragCurrent = new Point2D(mouseEvent.getScreenX(), mouseEvent.getScreenY());
+        Point2D dragCurrent = new Point2D(mouseEvent.getScreenX(),
+                mouseEvent.getScreenY());
```
Keep as one line or refactor.

TheMSB replied:
Fixed

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/ComponentLoader.java:
```diff
@@ -3,16 +3,17 @@
     default FXMLLoader getFXMLLoader(String name) {
-        FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource(String.format("/ui/%s.fxml", name)));
+        FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource(
+                String.format("/ui/%s.fxml", name)));
```
Keep as one line or refactor.

TheMSB replied:
Because of simplicity will let it stay as one line.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/anchors/ConnectionAnchor.java:
```diff
@@ -13,10 +14,13 @@
  */
-public abstract class ConnectionAnchor extends Circle implements ComponentLoader {
+public abstract class ConnectionAnchor extends Circle implements
+        ComponentLoader {
```
Keep as one line.

TheMSB replied:
On the fence about this one, will think about it a little more.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/anchors/ConnectionAnchor.java:
```diff
@@ -27,9 +31,12 @@
+     * @param pane
+     *            The pane this Anchor belongs to.
+     * @throws IOException
+     *             when the FXML definitions cannot be loaded.
```
I'm on the fence on this one.

TheMSB replied:
I believe this is the correct way, It results in properly formatted JavaDoc.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/anchors/ConnectionAnchor.java:
```diff
@@ -95,14 +93,30 @@ public boolean isConnected() {
     public String toString() {
-        return String.format("%s for %s", this.getClass().getSimpleName(), getBlock());
+        return String.format("%s for %s", this.getClass().getSimpleName(),
+                getBlock());
```
Keep as one line or refactor.

TheMSB replied:
Because of simplicity will let it stay as one line.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/anchors/InputAnchor.java:
```diff
@@ -15,9 +15,12 @@
+     * @param pane
+     *            The parent pane this Anchor resides on.
+     * @throws IOException
+     *             when the FXML definitions cannot be loaded.
```
See above.

TheMSB replied:
I believe this is the correct way, It results in properly formatted JavaDoc.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/blocks/FunctionBlock.java:
```diff
@@ -33,21 +33,33 @@
      */
-    public FunctionBlock(String name, Type type, CustomUIPane pane) throws IOException {
+    public FunctionBlock(String name, Type type, CustomUIPane pane)
+            throws IOException {
```
Keep as one line.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/blocks/FunctionBlock.java:
```diff
@@ -155,23 +137,51 @@ public final int getArgumentIndex(ConnectionAnchor anchor) {
         Expr expr = new Ident(getName());
-        for (InputAnchor in : getInputs()) expr = new Apply(expr, in.asExpr());
+        for (InputAnchor in : getInputs())
+            expr = new Apply(expr, in.asExpr());
```
If you're going to break this, add moustaches `{}`.

TheMSB replied:
Missed this one, added them.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/blocks/FunctionBlock.java:
```diff
@@ -155,23 +137,51 @@ public final int getArgumentIndex(ConnectionAnchor anchor) {
+        for (InputAnchor in : getInputs()) {
+            if (!in.isConnected()) {
+                argumentSpace.getChildren().get(getArgumentIndex(in))
+                        .getStyleClass().add("error");
```
Keep as one line or refactor.

TheMSB replied:
Refactored.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/blocks/FunctionBlock.java:
```diff
@@ -155,23 +137,51 @@ public final int getArgumentIndex(ConnectionAnchor anchor) {
+                        .getStyleClass().add("error");
+            } else if (in.isConnected()) {
+                argumentSpace.getChildren().get(getArgumentIndex(in))
+                        .getStyleClass().remove("error");
```
Keep as one line or refactor.

TheMSB replied:
Refactored.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/lines/Connection.java:
```diff
@@ -50,31 +56,23 @@ public Connection(InputAnchor to, OutputAnchor from) {
         boolean added = false;
-        if ((!startAnchor.isPresent() || override) && anchor instanceof OutputAnchor) {
+        if ((!startAnchor.isPresent() || override)
+                && anchor instanceof OutputAnchor) {
```
Keep as one line or refactor.

TheMSB replied:
Line break happens on && I believe since && is just a short-circuit for a nested if the usage here is obvious enough.

If you REALLY believe that it should either be one line or refactor then I will return it to one line.

wandernauta replied:
Yeah, it's obvious enough, but not really more obvious than the old version, I'd say.

How about something like this?

````
boolean added = false,
        isOutputAnchor = anchor instanceof OutputAnchor,
        isInputAnchor = anchor instanceof InputAnchor;

if (isOutputAnchor && (!startAnchor.isPresent() || override)) {
    disconnect(startAnchor);
...
````

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/lines/Connection.java:
```diff
@@ -50,31 +56,23 @@ public Connection(InputAnchor to, OutputAnchor from) {
             added = true;
-        } else if ((!endAnchor.isPresent() || override) && anchor instanceof InputAnchor) {
+        } else if ((!endAnchor.isPresent() || override)
+                && anchor instanceof InputAnchor) {
```
Keep as one line or refactor.

TheMSB replied:
See above.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/lines/Connection.java:
```diff
@@ -114,16 +92,20 @@ private void setAnchor(Optional anchor, ConnectionAnchor newAnchor) {
-                startAnchor.get().getBlock().layoutXProperty().removeListener(this);
-                startAnchor.get().getBlock().layoutYProperty().removeListener(this);
+                startAnchor.get().getBlock().layoutXProperty()
+                        .removeListener(this);
```
Keep as one line or refactor.

TheMSB replied:
Refactored.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/lines/Connection.java:
```diff
@@ -114,16 +92,20 @@ private void setAnchor(Optional anchor, ConnectionAnchor newAnchor) {
+                startAnchor.get().getBlock().layoutXProperty()
+                        .removeListener(this);
+                startAnchor.get().getBlock().layoutYProperty()
+                        .removeListener(this);
```
Keep as one line or refactor.

TheMSB replied:
Refactored.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/lines/Connection.java:
```diff
@@ -114,16 +92,20 @@ private void setAnchor(Optional anchor, ConnectionAnchor newAnchor) {
-                endAnchor.get().getBlock().layoutXProperty().removeListener(this);
-                endAnchor.get().getBlock().layoutYProperty().removeListener(this);
+                endAnchor.get().getBlock().layoutXProperty()
+                        .removeListener(this);
```
Keep as one line or refactor.

TheMSB replied:
Refactored.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/lines/Connection.java:
```diff
@@ -114,16 +92,20 @@ private void setAnchor(Optional anchor, ConnectionAnchor newAnchor) {
+                endAnchor.get().getBlock().layoutXProperty()
+                        .removeListener(this);
+                endAnchor.get().getBlock().layoutYProperty()
+                        .removeListener(this);
```
Keep as one line or refactor.

TheMSB replied:
Refactored.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/lines/Connection.java:
```diff
@@ -192,15 +206,17 @@ public String toString() {
+                // TODO Obviously this will cause errors, we need a way to
+                // access the Env
+                endAnchor.get().getBlock().asExpr()
+                        .analyze(new Env(), new GenSet());
```
Keep as one line or refactor.

TheMSB replied:
Refactored.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/lines/ConnectionLine.java:
```diff
@@ -46,8 +48,11 @@ public ConnectionLine() {
+     * @param x
+     *            coordinate
+     * @param y
+     *            coordinate
```
These `@param`s obviously fit on one line.

TheMSB replied:
While I believe the modified version is correct to present correctly formatted JavaDoc I changed the doc a little to make it fit on one line.

wandernauta replied:
Could you clarify what you mean by 'correctly formatted'?

I gave it a whirl and I didn't see any difference for the generated JavaDoc. The [standard](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#format) gives an example where it goes on one line. I understand that Eclipse autoformats `@param` tags to two lines, but I'm not sure that means it's the only correct way.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/components/lines/ConnectionLine.java:
```diff
@@ -61,8 +66,11 @@ public void setStartPosition(Point2D point) {
+     * @param x
+     *            coordinate
+     * @param y
+     *            coordinate
```
These `@param`s obviously fit on one line.

TheMSB replied:
See above.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/menu/MainMenu.java:
```diff
@@ -31,14 +32,15 @@ public MainMenu(HaskellCatalog catalog, CustomUIPane tactilePane) {
 
-            ArrayList<FunctionEntry> entries = new ArrayList<>(catalog.getCategory(category));
+            ArrayList<FunctionEntry> entries = new ArrayList<>(
+                    catalog.getCategory(category));
```
Keep as one line or refactor.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/menu/MainMenu.java:
```diff
@@ -57,12 +59,17 @@ public MainMenu(HaskellCatalog catalog, CustomUIPane tactilePane) {
 
-        this.getItems().addAll(valueBlockItem, displayBlockItem, sep, errorItem, quitItem);
+        this.getItems().addAll(valueBlockItem, displayBlockItem, sep,
+                errorItem, quitItem);
```
Keep as one line or refactor.

TheMSB replied:
Simple entry, will return it to one line.

wandernauta commented on Code/src/main/java/nl/utwente/group10/ui/menu/MainMenu.java:
```diff
@@ -57,12 +59,17 @@ public MainMenu(HaskellCatalog catalog, CustomUIPane tactilePane) {
+            // the Env here (don't just calculate it over and over). Or just
+            // pass the signature String.
+            FunctionBlock fb = new FunctionBlock(entry.getName(),
+                    entry.asHaskellObject(new Context()), parent);
```
Keep as one line or refactor.

TheMSB replied:
Refactored.

TheMSB commented:
So I think I went over all the feedback. I modified them as stated in the comments. One extra point: I encountered a method argument that wasn't being used at all in it's method so I removed it. Hope these changes are acceptable.

TheMSB commented:
Ignore my last commit for now, I included a file I shouldn't have.

TheMSB commented:
Should be fixed now, sorry for the trouble. Please check.

wandernauta commented:
Looks reasonable, will merge. 

Thanks! :sparkles:

TheMSB commented:
> Looks reasonable, will merge.
> 
> Thanks!

Thanks for the code review ^^
